src/components/utils/src/threads/posix_thread.cc

```diff
@@ -258,7 +262,7 @@ void Thread::stop() {
 
 void Thread::join() {
   LOG4CXX_AUTO_TRACE(logger_);
-  DCHECK(!pthread_equal(pthread_self(), handle_));
+  DCHECK(!IsCurrentThread());
```

Comment:
@anosach-luxoft, May be use `DCHECK_OR_RETUNR`?
In release we will get an leaked thread in stead of deadlock.